test/recipes/80-test_cmp_http_data/Mock/setup-mock.sh

```diff
@@ -0,0 +1,159 @@
+#! /bin/bash
+set -e
+#set -x  #for debug
+
+# This script generates the certificates needed for the CMP server and the signer
+
+if [[ ! -f $mkcert_sh ]]; then
+    mkcert_sh="../../../certs/mkcert.sh"
+fi
+
+# sever certificate algorithms
+if [ -z "$server_rootca_keyalg" ]; then
+    server_rootca_keyalg="MLDSA65"
+fi
+if [ -z "$server_leaf_keyalg" ]; then
+    server_leaf_keyalg="SLH-DSA-SHAKE-192s"
+fi
+# rootCACert update test case
+if [ -z "$new_rootca_keyalg" ]; then
+    new_rootca_keyalg="SLH-DSA-SHAKE-192s"
+fi
+
+# End-entity certificate algorithms
+if [ -z "$signer_rootca_keyalg" ]; then
+    signer_rootca_keyalg="MLDSA65"
+fi
+if [ -z "$signer_interca_keyalg" ]; then
+    signer_interca_keyalg="MLDSA65"
+fi
+if [ -z "$signer_subinterca_keyalg" ]; then
+    signer_subinterca_keyalg="MLDSA65"
+fi
+if [ -z "$signer_leaf_keyalg" ]; then
+    signer_leaf_keyalg="SLH-DSA-SHAKE-192s"
+fi
+
+# CMP server certificate
+rename_serverfiles() {
+    echo "Renaming server files"
+    rm server_root-key.pem server_root-pubkey.pem newWithNew-key.pem newWithNew-pubkey.pem
+    mv server_root-cert.pem server_root.crt
+    cp server_root.crt trusted.crt
+    if [[ -f big_trusted.crt ]]; then
+        cat trusted.crt >> big_trusted.crt
+    fi
+    cp server_root.crt oldWithOld.pem
+    mv server-key.pem server.key
+    mv server-cert.pem server.crt
+    mv newWithNew-cert.pem newWithNew.pem
+}
+
+remove_serverfiles() {
+    echo "Removing server files"
+    rm -f server.key server-crt trusted.crt server_root.crt
+    rm -f oldWithOld.pem newWithNew.pem oldWithNew.pem newWithOld.pem
+}
+
+gen_servercert() {
+    remove_serverfiles
+    sleep 5
+    OPENSSL_KEYALG=${server_rootca_keyalg} \
+    $mkcert_sh genroot "Root CA" server_root-key server_root-cert
+    OPENSSL_KEYALG=${server_leaf_keyalg} \
+    $mkcert_sh genee -p serverAuth,cmKGA server.example server-key server-cert server_root-key server_root-cert
+
+    OPENSSL_KEYALG=${new_rootca_keyalg} \
+    $mkcert_sh genroot "Root CA" newWithNew-key newWithNew-cert
+
+    openssl pkey -in newWithNew-key.pem -out newWithNew-pubkey.pem -outform PEM -pubout
+    openssl x509 -new -subj "/CN=Root CA" -CA server_root-cert.pem -CAkey server_root-key.pem \
+        -out newWithOld.pem -force_pubkey newWithNew-pubkey.pem -extensions SAN \
+        -extfile <(printf "[SAN]\nbasicConstraints=critical,CA:true")
```

Comment:
SAN is not the right name here, and such a section name is not needed.
```suggestion
        -out newWithOld.pem -force_pubkey newWithNew-pubkey.pem \
        -extfile <(printf "basicConstraints=critical,CA:true")
```
Also in the three further such cases below.